docs/rules/prefer-rest-params.md

```diff
@@ -43,6 +43,11 @@ function foo() {
     var arguments = 0;
     console.log(arguments); // This is a local variable.
 }
+
+// arguments can still be spread
+constructor() {
+  super(...arguments)
+}
```

Comment:
@mysticatea my understanding of the documentation of this rule, as you wrote it, is that its intention is to avoid confusing or potentially-hazardous accesses to `arguments`, namely using `Array` methods to transform `arguments`, or accessing its indices directly.

I suggest that spreading `arguments` has no possibility of encountering these issues-- the developer is treating `arguments` as opaque and is only passing it to another function (usually some sort of `super` in my experience).

My thought is that it's similar to accessing `arguments.length` where the developer is not concerned with trying to look into the arguments themselves. The rule does not force this behavior:

```
function(...args) {
  return args.length; // can just return arguments.length
}
```

So I would suggest that a spread falls into the same category.